### ISSUE TITLE
PCA is hanging and taking way too long

### ISSUE THREAD

**Issue by a-munoz-rojas** [scverse/scanpy#2531]
### Please make sure these conditions are met

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the latest version of scanpy.
- [ ] (optional) I have confirmed this bug exists on the master branch of scanpy.

### What happened?

I just made a new environment with the latest version of scanpy (1.9.3), and was trying to run some basic analysis and the pca function is hanging and taking forever, when it used to be basically instantaneous. I tried reproducing it with the the simple tutorial and it is still hanging when I try to run PCA (takes longer than 10 minutes on the simple example dataset in the tutorial, I eventually just have to kill the process). 

Not sure at all what the issue is, but wanted to bring it to your attention. This does not happen when using scanpy 1.9.1. Here is the output from the header logger:

scanpy==1.9.3 anndata==0.9.1 umap==0.5.3 numpy==1.24.3 scipy==1.9.1 pandas==2.0.2 scikit-learn==1.2.2 statsmodels==0.14.0 python-igraph==0.10.4 louvain==0.8.0 pynndescent==0.5.10

### Minimal code sample

```python
#Follow the same as the tutorial, or as a minimum example:

import numpy as np
import pandas as pd
import scanpy as sc

adata = sc.datasets.pbmc3k()
adata.var_names_make_unique() 
sc.pp.filter_cells(adata, min_genes=200)
sc.pp.filter_genes(adata, min_cells=3)
adata.var['mt'] = adata.var_names.str.startswith('MT-')
sc.pp.calculate_qc_metrics(adata, qc_vars=['mt'], percent_top=None, log1p=False, inplace=True)
adata = adata[adata.obs.n_genes_by_counts < 2500, :]
adata = adata[adata.obs.pct_counts_mt < 5, :]
sc.pp.normalize_total(adata, target_sum=1e4)
sc.pp.log1p(adata)
sc.pp.highly_variable_genes(adata, min_mean=0.0125, max_mean=3, min_disp=0.5)
sc.tl.pca(adata, svd_solver='arpack')
```

### Error output
No error output, it just hangs
```pytb

```

### Versions

<details>

```
-----
anndata     0.9.1
scanpy      1.9.3
-----
CoreFoundation      NA
Foundation          NA
PIL                 9.5.0
PyObjCTools         NA
anyio               NA
appnope             0.1.3
argcomplete         NA
asttokens           NA
attr                23.1.0
babel               2.12.1
backcall            0.2.0
beta_ufunc          NA
binom_ufunc         NA
certifi             2023.05.07
cffi                1.15.1
charset_normalizer  3.1.0
cloudpickle         2.2.1
colorama            0.4.6
comm                0.1.3
cycler              0.10.0
cython_runtime      NA
cytoolz             0.12.0
dask                2023.6.0
dateutil            2.8.2
debugpy             1.6.7
decorator           5.1.1
defusedxml          0.7.1
executing           1.2.0
fastjsonschema      NA
google              NA
h5py                3.9.0
hypergeom_ufunc     NA
idna                3.4
igraph              0.10.4
importlib_resources NA
ipykernel           6.23.2
ipython_genutils    0.2.0
ipywidgets          8.0.6
jedi                0.18.2
jinja2              3.1.2
joblib              1.2.0
json5               NA
jsonschema          4.17.3
jupyter_events      0.6.3
jupyter_server      2.6.0
jupyterlab_server   2.23.0
kiwisolver          1.4.4
leidenalg           0.9.1
llvmlite            0.40.0
louvain             0.8.0
markupsafe          2.1.3
matplotlib          3.7.1
mpl_toolkits        NA
natsort             8.4.0
nbformat            5.9.0
nbinom_ufunc        NA
ncf_ufunc           NA
numba               0.57.0
numexpr             2.8.4
numpy               1.24.3
objc                9.2
overrides           NA
packaging           23.1
pandas              2.0.2
parso               0.8.3
pexpect             4.8.0
pickleshare         0.7.5
pkg_resources       NA
platformdirs        3.6.0
prometheus_client   NA
prompt_toolkit      3.0.38
psutil              5.9.5
ptyprocess          0.7.0
pure_eval           0.2.2
pvectorc            NA
pydev_ipython       NA
pydevconsole        NA
pydevd              2.9.5
pydevd_file_utils   NA
pydevd_plugins      NA
pydevd_tracing      NA
pygments            2.15.1
pyparsing           3.1.0
pyrsistent          NA
pythonjsonlogger    NA
pytz                2023.3
requests            2.31.0
rfc3339_validator   0.1.4
rfc3986_validator   0.1.1
scipy               1.9.1
send2trash          NA
session_info        1.0.0
setuptools          68.0.0
six                 1.16.0
sklearn             1.2.2
sniffio             1.3.0
socks               1.7.1
stack_data          0.6.2
texttable           1.6.7
threadpoolctl       3.1.0
tlz                 0.12.0
toolz               0.12.0
torch               1.12.1
tornado             6.3.2
tqdm                4.65.0
traitlets           5.9.0
typing_extensions   NA
urllib3             2.0.3
wcwidth             0.2.6
websocket           1.6.0
yaml                6.0
zipp                NA
zmq                 25.1.0
zoneinfo            NA
-----
IPython             8.14.0
jupyter_client      8.2.0
jupyter_core        5.3.1
jupyterlab          4.0.2
notebook            6.5.4
-----
Python 3.9.16 | packaged by conda-forge | (main, Feb  1 2023, 21:42:20) [Clang 14.0.6 ]
macOS-12.6.6-x86_64-i386-64bit
-----
Session information updated at 2023-06-23 18:00

```

</details>

**Comment by flying-sheep** [comment]
Hm, I can’t reproduce this, but I also can’t reproduce your environment, since scipy 1.9.1 doesn’t build on my system.

With this environment, everything worked instantly. Can you try updating your environment?

```
anndata==0.9.1
asttokens==2.2.1
backcall==0.2.0
contourpy==1.1.0
cycler==0.11.0
decorator==5.1.1
executing==1.2.0
fonttools==4.40.0
h5py==3.9.0
igraph==0.10.4
ipython==8.14.0
jedi==0.18.2
joblib==1.2.0
kiwisolver==1.4.4
llvmlite==0.40.1
louvain==0.8.0
matplotlib==3.7.1
matplotlib-inline==0.1.6
natsort==8.4.0
networkx==3.1
numba==0.57.1
numpy==1.24.3
packaging==23.1
pandas==2.0.2
parso==0.8.3
patsy==0.5.3
pexpect==4.8.0
pickleshare==0.7.5
Pillow==9.5.0
prompt-toolkit==3.0.38
ptyprocess==0.7.0
pure-eval==0.2.2
Pygments==2.15.1
pynndescent==0.5.10
pyparsing==3.1.0
python-dateutil==2.8.2
python-igraph==0.10.4
pytz==2023.3
scanpy==1.9.3
scikit-learn==1.2.2
scipy==1.11.0
seaborn==0.12.2
session-info==1.0.0
six==1.16.0
stack-data==0.6.2
statsmodels==0.14.0
stdlib-list==0.9.0
texttable==1.6.7
threadpoolctl==3.1.0
tqdm==4.65.0
traitlets==5.9.0
tzdata==2023.3
umap-learn==0.5.3
wcwidth==0.2.6
```

**Comment by a-munoz-rojas** [comment]
Thanks for getting back to me. I tried updating the environment (spicy now at 1.11.0), and it's still hanging during the PCA step. Here are the updated versions:

### Versions

<details>

```
-----
anndata     0.9.1
scanpy      1.9.3
-----
CoreFoundation      NA
Foundation          NA
PIL                 9.5.0
PyObjCTools         NA
anyio               NA
appnope             0.1.3
argcomplete         NA
asttokens           NA
attr                23.1.0
babel               2.12.1
backcall            0.2.0
certifi             2023.05.07
cffi                1.15.1
charset_normalizer  3.1.0
cloudpickle         2.2.1
colorama            0.4.6
comm                0.1.3
cycler              0.10.0
cython_runtime      NA
cytoolz             0.12.0
dask                2023.6.0
dateutil            2.8.2
debugpy             1.6.7
decorator           5.1.1
defusedxml          0.7.1
executing           1.2.0
fastjsonschema      NA
google              NA
h5py                3.9.0
idna                3.4
igraph              0.10.4
importlib_resources NA
ipykernel           6.23.3
ipython_genutils    0.2.0
ipywidgets          8.0.6
jedi                0.18.2
jinja2              3.1.2
joblib              1.2.0
json5               NA
jsonschema          4.17.3
jupyter_events      0.6.3
jupyter_server      2.6.0
jupyterlab_server   2.23.0
kiwisolver          1.4.4
leidenalg           0.9.1
llvmlite            0.40.0
louvain             0.8.0
markupsafe          2.1.3
matplotlib          3.7.1
mpl_toolkits        NA
natsort             8.4.0
nbformat            5.9.0
numba               0.57.0
numexpr             2.8.4
numpy               1.24.3
objc                9.2
overrides           NA
packaging           23.1
pandas              2.0.2
parso               0.8.3
patsy               0.5.3
pexpect             4.8.0
pickleshare         0.7.5
pkg_resources       NA
platformdirs        3.8.0
prometheus_client   NA
prompt_toolkit      3.0.38
psutil              5.9.5
ptyprocess          0.7.0
pure_eval           0.2.2
pvectorc            NA
pydev_jupyter_utils NA
pydev_jupyter_vars  NA
pydevconsole        NA
pydevd_file_utils   NA
pydevd_plugins      NA
pygments            2.15.1
pyparsing           3.1.0
pyrsistent          NA
pythonjsonlogger    NA
pytz                2023.3
requests            2.31.0
rfc3339_validator   0.1.4
rfc3986_validator   0.1.1
scipy               1.11.0
seaborn             0.12.2
send2trash          NA
session_info        1.0.0
setuptools          68.0.0
six                 1.16.0
sklearn             1.2.2
sniffio             1.3.0
socks               1.7.1
stack_data          0.6.2
statsmodels         0.14.0
texttable           1.6.7
threadpoolctl       3.1.0
tlz                 0.12.0
toolz               0.12.0
torch               1.12.1
tornado             6.3.2
tqdm                4.65.0
traitlets           5.9.0
typing_extensions   NA
urllib3             2.0.3
wcwidth             0.2.6
websocket           1.6.1
yaml                6.0
zipp                NA
zmq                 25.1.0
zoneinfo            NA
-----
IPython             8.14.0
jupyter_client      8.3.0
jupyter_core        5.3.1
jupyterlab          4.0.2
notebook            6.5.4
-----
Python 3.9.16 | packaged by conda-forge | (main, Feb  1 2023, 21:42:20) [Clang 14.0.6 ]
macOS-12.6.6-x86_64-i386-64bit
-----
Session information updated at 2023-06-26 14:43

```

</details>

**Comment by flying-sheep** [comment]
Hm, PyTorch 1.12.1 doesn’t seem to be available, otherwise I was able to install all package versions you specified.

I created an attempt at a reproducer here, but it just runs in less than a second for me: https://github.com/flying-sheep/scanpy-2531

No idea how to further debug this. Looks like you’re on macOS. one of the new M1 or M2 ones?

**Comment by a-munoz-rojas** [comment]
Thanks for getting back. I ran the reproducer on my system and it indeed works perfectly. Very weird. 

When I ran the reproducer, I did get one warning:
```
WARNING: The candidate selected for download or install is a yanked version: 'scipy' candidate (version 1.11.0 at https://files.pythonhosted.org/packages/2f/b5/b5387cdafc66805907424c3a95f773b84a5d452a0925801c6218727a766e/scipy-1.11.0-cp311-cp311-macosx_10_9_x86_64.whl (from https://pypi.org/simple/scipy/) (requires-python:<3.13,>=3.9))
Reason for being yanked: License Violation
```
Other than that, it worked fine. I have a feeling it might be an issue with the installation? That scipy warning is suspicious? I'm using mamba (mambaforge specifically) to manage my packages, maybe something went wrong there. Have you hear of any issues with mamba? Let me trouble shoot my environment and I'll report back. 

Yes, I am running macOS, but it's not the Apple Silicon, I'm still using the older Intel processors.
